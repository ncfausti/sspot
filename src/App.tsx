import React from 'react';
import { HashRouter as Router, Switch, Route } from 'react-router-dom';
import './App.global.css';
// import log from 'electron-log';
// import { AuthProvider } from './contexts/AuthContext';
// import PrivateRoute from './components/PrivateRoute';
import Meetings from './components/Meetings';
import Hud from './components/Hud';
// import SignIn from './components/SignIn';

export default function App() {
  return (
    <Router>
      {/* <AuthProvider> */}
      <Switch>
        <Route exact path="/" component={Meetings} />

        {/* <Route path="/login" component={SignIn} /> */}
        <Route path="/live" component={Hud} />
      </Switch>
      {/* </AuthProvider> */}
    </Router>
  );
}
