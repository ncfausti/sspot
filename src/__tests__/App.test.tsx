import '@testing-library/jest-dom';
import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

// Mock electron, electron-store, and DataStore
jest.mock('electron', () => {
  const mElectron = {
    ipcRenderer: {
      on: jest.fn(),
      send: jest.fn(),
      sendSync: jest.fn(),
    },
    remote: { getGlobal: jest.fn() },
  };
  return mElectron;
});

// jest.mock('remote');

// Partial mock example
// https://jestjs.io/docs/mock-functions#mocking-partials
// jest.mock('../DataStore', () => {
//   const originalModule = jest.requireActual('../DataStore');

//   // Mock the default export and named export Collection
//   return {
//     __esModule: true,
//     Collection: [],
//     ...originalModule,
//     default: () => {
//       return { getAll: () => [] };
//     },
//   };
// });

describe('App', () => {
  it('should render', () => {
    expect(render(<App />)).toBeTruthy();
  });

  it('should render with snapshot', () => {
    expect(render(<App />)).toMatchSnapshot();
  });
});
