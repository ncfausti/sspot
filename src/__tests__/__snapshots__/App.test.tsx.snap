// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex bg-gray-100 flex-grow flex-col p-3 min-h-screen content-center"
      >
        <div
          class="flex flex-grow flex-wrap justify-between"
        >
          <div
            class="text-xs text-gray-800 font-semibold"
          >
            December 18, 2021
          </div>
          <div
            class="text-xs font-light"
          >
            11:32 PM CST
          </div>
        </div>
        <div
          class="flex flex-grow"
        >
          <button
            class="w-full border border-transparent text-xs font-semibold font-medium rounded-md shadow-sm text-white bg-blue-500 hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-400"
            type="button"
          >
            Launch SaleSpot
          </button>
        </div>
        <div
          class="flex flex-grow flex-wrap justify-between items-center"
        >
          <div
            class=""
          >
            <img
              alt="SaleSpot"
              class="inline w-16 mr-1"
              src="test-file-stub"
            />
            <span
              class="inline text-xs text-gray-400 font-light"
              style="font-size: 8px;"
            >
              ⌘ + Y to toggle
            </span>
          </div>
          <div
            class="text-gray-700"
          >
            <svg
              aria-hidden="true"
              class="h-4 w-4 cursor-pointer"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.49 3.17c-.38-1.56-2.6-1.56-2.98 0a1.532 1.532 0 01-2.286.948c-1.372-.836-2.942.734-2.106 2.106.54.886.061 2.042-.947 2.287-1.561.379-1.561 2.6 0 2.978a1.532 1.532 0 01.947 2.287c-.836 1.372.734 2.942 2.106 2.106a1.532 1.532 0 012.287.947c.379 1.561 2.6 1.561 2.978 0a1.533 1.533 0 012.287-.947c1.372.836 2.942-.734 2.106-2.106a1.533 1.533 0 01.947-2.287c1.561-.379 1.561-2.6 0-2.978a1.532 1.532 0 01-.947-2.287c.836-1.372-.734-2.942-2.106-2.106a1.532 1.532 0 01-2.287-.947zM10 13a3 3 0 100-6 3 3 0 000 6z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex bg-gray-100 flex-grow flex-col p-3 min-h-screen content-center"
    >
      <div
        class="flex flex-grow flex-wrap justify-between"
      >
        <div
          class="text-xs text-gray-800 font-semibold"
        >
          December 18, 2021
        </div>
        <div
          class="text-xs font-light"
        >
          11:32 PM CST
        </div>
      </div>
      <div
        class="flex flex-grow"
      >
        <button
          class="w-full border border-transparent text-xs font-semibold font-medium rounded-md shadow-sm text-white bg-blue-500 hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-400"
          type="button"
        >
          Launch SaleSpot
        </button>
      </div>
      <div
        class="flex flex-grow flex-wrap justify-between items-center"
      >
        <div
          class=""
        >
          <img
            alt="SaleSpot"
            class="inline w-16 mr-1"
            src="test-file-stub"
          />
          <span
            class="inline text-xs text-gray-400 font-light"
            style="font-size: 8px;"
          >
            ⌘ + Y to toggle
          </span>
        </div>
        <div
          class="text-gray-700"
        >
          <svg
            aria-hidden="true"
            class="h-4 w-4 cursor-pointer"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.49 3.17c-.38-1.56-2.6-1.56-2.98 0a1.532 1.532 0 01-2.286.948c-1.372-.836-2.942.734-2.106 2.106.54.886.061 2.042-.947 2.287-1.561.379-1.561 2.6 0 2.978a1.532 1.532 0 01.947 2.287c-.836 1.372.734 2.942 2.106 2.106a1.532 1.532 0 012.287.947c.379 1.561 2.6 1.561 2.978 0a1.533 1.533 0 012.287-.947c1.372.836 2.942-.734 2.106-2.106a1.533 1.533 0 01.947-2.287c1.561-.379 1.561-2.6 0-2.978a1.532 1.532 0 01-.947-2.287c.836-1.372-.734-2.942-2.106-2.106a1.532 1.532 0 01-2.287-.947zM10 13a3 3 0 100-6 3 3 0 000 6z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
